File: orientedDisign/library/borrow_transaction.py
from dataclasses import dataclass
from datetime import datetime


@dataclass
class BorrowTransaction:
    transaction_id: str
    user_id: str
    book_id: str
    borrowed_at: datetime
    returned: bool = False
    overdue: bool = False

    def mark_returned(self):
        self.returned = True

    def mark_overdue(self):
        self.overdue = True

File: orientedDisign/js_demo/borrowTransaction.js
class BorrowTransaction {
  constructor(transactionId, userId, bookId) {
    this.transactionId = transactionId;
    this.userId = userId;
    this.bookId = bookId;
    this.borrowedAt = new Date();
    this.returned = false;
    this.overdue = false;
  }

  markReturned() {
    this.returned = true;
  }

  markOverdue() {
    this.overdue = true;
  }
}

module.exports = BorrowTransaction;
